ted/src/main/java/frc/robot/commands/ShooterSetAngleUntilLimitCommand.java

```diff
@@ -0,0 +1,136 @@
+  private static final double secondStepSizeDegrees = firstStepSizeDegrees * 2.0;
+  private static final double thirdTimeLimitSeconds = 1.4;
+  private static final double thirdStepSizeDegrees = secondStepSizeDegrees * 1.75;
+  private static final double fourthTimeLimitSeconds = 1.8;
+  private static final double fourthStepSizeDegrees = firstStepSizeDegrees * 1.5;
+
+  private ShooterAngleSubsystem shooter;
+  private boolean increasingAngle = false;
+  private Timer timer = new Timer();
+  private boolean done = false;
+
+  /**
+   * Constructor for ShooterSetAngleTesterCommand
+   * Will update the shooter angle in one direction until the directional limit is reached (maximum limit for the 
+   * increasing direction and minimum limit for decreasing) or until the command is interrupted by the scheduler 
+   * @param increaseAngle - When true the shooter angle will increase until either the maximum limit is reached or 
+   * @param shooter - the shooter angle subsystem in use currently
+   */
+  public ShooterSetAngleUntilLimitCommand(boolean increaseAngle, ShooterAngleSubsystem shooter) {
+    this.increasingAngle = increaseAngle;
+    this.shooter = shooter;
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    done = false;
+    timer.reset();
+    timer.start();
+  }
+
+  // buisness logic to move the shooter once per robot tick (every 20ms / 50Hz)
+  @Override
+  public void execute() {
+
+    double angleStepSize = this.getAngleStepSize();
+    double currentShooterAngle = this.shooter.getAngleDegrees();
+    double proposedAngle = currentShooterAngle;
+
+    if(increasingAngle ) {
+        if(currentShooterAngle >= Constants.shooterAngleMaxDegrees) {
+            done = true;
+        }
+        else {
+            proposedAngle = currentShooterAngle + angleStepSize;
+        }
+    }
+    else {
+        if(currentShooterAngle <= Constants.shooterAngleMinDegrees) {
+            done = true;
+        }
+        else {
+            proposedAngle = currentShooterAngle - angleStepSize;
+        }
+    }
+
+    if(!done) {
+        shooter.setAngleDegrees(
+            MotorUtils.clamp(
+                proposedAngle,
+                Constants.shooterAngleMinDegrees,
+                Constants.shooterAngleMaxDegrees));
+    }
+  }
+ 
+  // Called once the command ends or is interrupted.
+  @Override
+  public void end(boolean interrupted) {
+    if(interrupted){
+      done = true;
+    }
+  }
+
+  // Returns true when the command should end.
+  @Override
+  public boolean isFinished() {
+    return done;
+  }
+
+  /**
+   * A method to return the current angle step size based on how long the command has
+   * been running
+   * @return double representing the angle step size in degrees
+   */
+  private double getAngleStepSize() {
+    double elapsedTimeSeconds = timer.get();
+    double angleStepSize = Constants.shooterAngleStickIncrementMagnitude;
```

Comment:
It seems like it will cause confusion to have one of the speed values in Constants.java, but the rest of the speed values declared here in this command.  Someone who thinks the stick is not moving fast enough would change the value in constants, but that would only affect a small part of the behavior of this command. I'm wondering if all the speeds should be a multiple of the one in constants? That would allow one change in constants to "speed up" the entire behavior of this command. Another option is to have all the speed values declared in this file so that when someone goes looking, they will find them all in one place.

Reply:
All increment values ultimately root into Constants.shooterAngleStickIncrementMagnitude.

Reply:
we went with a single setting here ... so it seems this is done